src/preprocess.py

```diff
@@ -262,14 +285,15 @@ class InvertedIndex:
     """
     itype: int
     index: Union[
-        OrdDict[str, List[int]], # itype == 1
+        OrdDict[str, OrdDict[int, bool]], # itype == 1
```

Comment:
I guess the ultimate optimization for the inner data structure would be a numpy array given the number of documents. Given the size of the vocabulary of the collection, you can even have a 2D array where rows are ordered by alphabetic order on the vocab. But hey if it works, that's fine with me.